src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -878,23 +946,73 @@ class ExposureSubmissionCoordinator: NSObject, RequiresAppDependencies {
+	/// - Parameter isSelfTestTypePreselected: If the process was started via self-report with self-test on the Manage Your Tests View, the Self-test entry shall be pre-selected. Otherwise, no entry shall be selected.
+	private func showSRSTestTypeSelectionScreen(isSelfTestTypePreselected: Bool = false) {
+		let srsTestTypeSelectionViewController = SRSTestTypeSelectionViewController(
+			viewModel: SRSTestTypeSelectionViewModel(isSelfTestTypePreselected: isSelfTestTypePreselected),
+			onPrimaryButtonTap: { [weak self] submissionType in
+				self?.model.storeSelectedSRSSubmissionType(submissionType)
+				self?.showSRSFlowNextScreen()
+			}, onDismiss: { [weak self] in
+				self?.parentViewController?.dismiss(animated: true)
+			}
+		)
+		
+		let footerViewController = FooterViewController(
+			FooterViewModel(
+				primaryButtonName: AppStrings.ExposureSubmission.SRSTestTypeSelection.primaryButtonTitle,
+				isSecondaryButtonEnabled: false,
+				isSecondaryButtonHidden: true
+			)
+		)
+		
+		let topBottomContainerViewController = TopBottomContainerViewController(
+			topController: srsTestTypeSelectionViewController,
+			bottomController: footerViewController
+		)
+
+		push(topBottomContainerViewController)
+	}
+	
+	private func showSRSFlowNextScreen() {
+		if model.eventProvider.checkinsPublisher.value.isEmpty {
+			showSymptomsScreen()
+		} else {
+			showCheckinsScreen(isSRSFlow: true)
+		}
+	}
+
 	// MARK: Symptoms
 
 	private func showSymptomsScreen() {
-		if let testType = model.coronaTestType {
-			Analytics.collect(.keySubmissionMetadata(.lastSubmissionFlowScreen(.submissionFlowScreenSymptoms, testType)))
+		if let coronaTestType = model.coronaTestType {
+			Analytics.collect(.keySubmissionMetadata(.lastSubmissionFlowScreen(.submissionFlowScreenSymptoms, coronaTestType)))
 		}
 
-		let vc = ExposureSubmissionSymptomsViewController(
+		let exposureSubmissionSymptomsViewController = ExposureSubmissionSymptomsViewController(
 			onPrimaryButtonTap: { [weak self] selectedSymptomsOption, isLoading in
 				guard let self = self else { return }
 
 				self.model.symptomsOptionSelected(selectedSymptomsOption)
-				// we don't need to set it true if yes is selected
-				if selectedSymptomsOption != .yes, let testType = self.model.coronaTestType {
-					Analytics.collect(.keySubmissionMetadata(.submittedAfterSymptomFlow(true, testType)))
+				
+				switch self.model.submissionTestType {
+				case .registeredTest(let coronaTestType):
+					// we don't need to set it true if yes is selected
+					if selectedSymptomsOption != .yes, let coronaTestType = coronaTestType {
+						Analytics.collect(.keySubmissionMetadata(.submittedAfterSymptomFlow(true, coronaTestType)))
+					}
+					
+					self.model.shouldShowSymptomsOnsetScreen ? self.showSymptomsOnsetScreen() : self.submitExposure(showSubmissionSuccess: true, isLoading: isLoading)
+				case .srs:
+					if self.model.shouldShowSymptomsOnsetScreen {
+						self.showSymptomsOnsetScreen()
```

Comment:
We submit on symptomsOnsetScreen not here

Reply:
Sorry, I was not fully clear in my comment: I meant the submission is missing in case `model.shouldShowSymptomsOnsetScreen` is false, e.g. if the user selected that they have no symptoms. I think the logic should be the same as for the "normal" submission above

Reply:
Fixed with https://github.com/corona-warn-app/cwa-app-ios/pull/4900